test/stancjs/stancjs.expected

```diff
@@ -205,6 +205,69 @@ Syntax error in 'string', line 8, column 0 to column 5, parsing error:
 
 Expected "generated quantities {" or end of file after end of model block.
 
+$ node includes.js
+Syntax error in 'string', line 2, column 0, include error:
+   -------------------------------------------------
+     1:  
+     2:  #include <foo.stan>
+         ^
+     3:  data {
+     4:      int a;
+   -------------------------------------------------
+
+Could not find include file 'foo.stan'.
+stanc was given information about the following files:
+None
+
+Syntax error in 'string', line 2, column 0, include error:
+   -------------------------------------------------
+     1:  
+     2:  #include <foo.stan>
+         ^
+     3:  data {
+     4:      int a;
+   -------------------------------------------------
+
+Could not find include file 'foo.stan'.
+stanc was given information about the following files:
+bar.stan
+
+functions {
+  int foo(real a) {
+    return 1;
+  }
+}
+data {
+  int a;
+}
+
+{
+  "inputs": { "a": { "type": "int", "dimensions": 0 } },
+  "parameters": {},
+  "transformed parameters": {},
+  "generated quantities": {},
+  "functions": [],
+  "distributions": [],
+  "included_files": [ "foo.stan" ]
+}
+Warning: stanc.js failed to parse included file mapping!
+Failed to read property 'foo.stan' of included files map!
+It had type 'object' instead of 'string'.
+Warning: stanc.js failed to parse included file mapping!
+Duplicate string in included files map: 'foo.stan'
+Warning: stanc.js failed to parse included file mapping!
+Included files map was provided but was not of type 'object'
+Syntax error in 'include/a.stan', line 1, column 0, included from
+'include/b.stan', line 1, column 0, included from
+'include/a.stan', line 1, column 0, included from
+'string', line 1, column 0, include error:
+   -------------------------------------------------
+     1:  #include <include/a.stan>
+         ^
+   -------------------------------------------------
+
+File include/b.stan recursively included itself.
```

Comment:
Errors.pp doesn't need it, this is because of the lack of the `filename-in-msg` argument. I have thought a couple times about making that argument essentially required in stancjs, because it's almost always wrong for a caller to not specify it. I've updated the test to read better.

Reply:
You misunderstood the issue. (This recursive include was a poor example of the general pattern.) Change the includes in the test to
```js
var recursive_a = `// comment here\n#include <include/b.stan>`
var recursive_b = `#include <include/a.stan>\n // comment here`
```
and you'll see the caret points to nonsense.

`Errors.pp` calls `Errors.pp_context_with_message` which reads from the filesystem. It would be possible for whoever calls `Errors.pp` to check the filename and supply the correct file, but that's not what currently happens, and in my opinion it we should keep the file-finding logic inside `Errors.pp_context_with_message`.

Reply:
Ah, thanks for that example.

Let me know what you think of my latest commit. To break a circular dependency I needed to move the ref cell out of `Preprocessor`, but I think the result is decent.